### PR TITLE
[CARBONDATA-1851] Refactor to use only SegmentsToAccess for Aggregatetable, move tableFolderDeletion to TableProcessingOperations 

### PR DESCRIPTION
**Pull request by rahulforallp**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed? **user interface is not changed.**
 
 - [X] Any backward compatibility impacted? **No**
 
 - [X] Document update required? **No**

 - [X] Testing done
        This PR contains **refactored code**  . So no new test-case required.
       Only UT, FT and SDV test success report required from CI is sufficient.

 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 

**1) CarbonTableInputFormat: Removed AgregateSegmentsToAccess interface as SegmentsToAccess Interface already does the same job.
Fix NullPointerException when SegmentsToAccess is configured and validateSegments flag is set false.
2) moved tablefolderdeletion logic to TableProcessingOperations to make those APIs developerAPI
3) Added a setter to CarbonProperties to add extra properties
4) Added a getter getImplicitDimensionByTableName**

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/466/

File: core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java [hunk 1]
```diff
@@ -121,6 +121,8 @@
 
   private boolean hasDataMapSchema;
 
+  private Map<String, Object> propertiesMap;
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java]: There is a table property map in the TableSchema already, it is in the TableInfo

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/471/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1735/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2114/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2119/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/505/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/508/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1766/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1770/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2147/

**Comment by rahulforallp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/546/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1803/

File: core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java [hunk 2]
```diff
@@ -990,4 +990,11 @@ public boolean isCarbonProperty(String key) {
     return addedProperty;
   }
 
+  /**
+   *
+   * @param externalProperties
+   */
+  public void addPropertyToPropertSet(Set<String> externalProperties) {
```
Comment (gvramana) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonProperties.java]: 'y' missing

File: hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java [hunk 3]
```diff
@@ -126,6 +126,10 @@
   private static final String DATA_MAP_DSTR = "mapreduce.input.carboninputformat.datamapdstr";
   public static final String DATABASE_NAME = "mapreduce.input.carboninputformat.databaseName";
   public static final String TABLE_NAME = "mapreduce.input.carboninputformat.tableName";
+  // to be used in cases where main table status file is not done and newly created segment need to
+  // be accessed
+  public static final String FORCE_ACCESS_SEGMENT =
+      "mapreduce.input.carboninputformat.force.access.segment";
```
Comment (gvramana) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonTableInputFormat.java]: Dont needs a new flag, use getValidateSegmentsToAccess, mapreduce.input.carboninputformat.validsegments

File: integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala [hunk 4]
```diff
@@ -151,58 +136,76 @@ object CarbonSession {
...
+              sc.conf.setAppName(randomAppName)
+            }
+            sc
           }
-        })
-        session.streams.addListener(new CarbonStreamingQueryListener(session))
-      }
 
-      session
+          session = new CarbonSession(sparkContext)
```
Comment (gvramana) [inline integration/spark2/src/main/scala/org/apache/spark/sql/CarbonSession.scala]: Why these changes

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java [hunk 5]
```diff
@@ -132,10 +133,25 @@ public boolean accept(CarbonFile file) {
   public static void deletePartialLoadDataIfExist(CarbonLoadModel loadModel,
       final boolean isCompactionFlow) throws IOException {
     CarbonTable carbonTable = loadModel.getCarbonDataLoadSchema().getCarbonTable();
-    String metaDataLocation = carbonTable.getMetaDataFilepath();
-    final LoadMetadataDetails[] details = SegmentStatusManager.readLoadMetadata(metaDataLocation);
+    deletePartialLoad(loadModel, carbonTable, isCompactionFlow);
+  }
+
+  public static void deletePartialLoad(CarbonLoadModel loadModel, CarbonTable carbonTable,
```
Comment (gvramana) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java]: Use same name, overloaded function deletePartialLoadDataIfExist

File: processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java [hunk 6]
```diff
@@ -196,8 +212,13 @@ public static void deleteStorePath(String path) {
    */
   public static void deleteLocalDataLoadFolderLocation(CarbonLoadModel loadModel,
       boolean isCompactionFlow, boolean isAltPartitionFlow) {
-    String databaseName = loadModel.getDatabaseName();
     String tableName = loadModel.getTableName();
+    deleteLocalDataLoadFolderLocation(loadModel, isCompactionFlow, isAltPartitionFlow, tableName);
```
Comment (gvramana) [inline processing/src/main/java/org/apache/carbondata/processing/util/CarbonLoaderUtil.java]: Define TableProcessingOperations Class, mark it as developer api, so that it is not modified.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/582/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/584/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1822/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/596/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1826/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2195/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2196/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/599/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1829/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1832/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2201/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/604/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1835/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1837/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/609/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1838/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/610/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/613/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1841/

**Comment by rahulforallp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/625/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1853/

**Comment by rahulforallp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/630/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1859/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/632/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1861/

**Comment by rahulforallp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/635/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1864/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2214/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/657/

**Comment by gvramana** [comment]
LGTM

**Comment by gvramana** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1888/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/659/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/661/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1890/
